src/main/java/de/tum/in/www1/artemis/config/migration/entries/BambooMigrationService.java

```diff
@@ -0,0 +1,463 @@
+
+        restTemplate.exchange(builder.build().toUri(), HttpMethod.POST, null, String.class);
+    }
+
+    /**
+     * Returns the id of the shared credential for the git user configured in the application properties.
+     *
+     * @return the id of the shared credential for the git user configured in the application properties
+     */
+    private Optional<Long> getSharedCredential() {
+        String requestUrl = bambooServerUrl + "/admin/credentials/configureSharedCredentials.action";
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(requestUrl);
+
+        var response = restTemplate.exchange(builder.build().toUri(), HttpMethod.GET, null, String.class);
+        var html = response.getBody();
+        if (html == null) {
+            return Optional.empty();
+        }
+        Elements entries = Jsoup.parse(html).select("table#sharedCredentialsTable tbody");
+
+        for (Element entry : entries) {
+            Elements row = entry.select("tr");
+            Element name = entry.selectFirst("td");
+            if (name != null && name.text().equals(gitUser)) {
+                var id = row.attr("id");
+                return Optional.of(Long.parseLong(id.replace("item-", "")));
+            }
+
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Adds a new git repository to the given build plan. The repository is configured to use the given credentials.
+     * Bamboo doesn't provide a REST endpoint for this action, so we have to directly make the requests.
+     *
+     * @param buildPlanKey  The key of the build plan, e.g. 'EIST16W1-BASE'.
+     * @param repository    The URL of the repository
+     * @param name          The name of the repository, e.g. 'assignment' or 'tests'
+     * @param credentialsId The id of the credentials to use for the repository
+     */
+    private void addGitRepository(String buildPlanKey, String repository, String name, Long credentialsId) {
+        MultiValueMap<String, String> body = new LinkedMultiValueMap<>();
+        body.add("planKey", buildPlanKey);
+        body.add("repositoryId", Integer.toString(0));
+        body.add("selectedRepository", "com.atlassian.bamboo.plugins.atlassian-bamboo-plugin-git:gitv2");
+        body.add("respositoryPluginKey", "com.atlassian.bamboo.plugins.atlassian-bamboo-plugin-git:gitv2");
+        body.add("repositoryName", name);
+        body.add("repository.git.repositoryUrl", repository);
+        body.add("repository.git.authenticationType", "PASSWORD");
+        body.add("selectFields", "repository.git.authenticationType");
+        body.add("repository.git.passwordCredentialsSource", "SHARED_CREDENTIALS");
+        body.add("repository.git.passwordSharedCredentials", credentialsId.toString());
+        body.add("selectFields", "repository.git.passwordSharedCredentials");
+        body.add("repository.git.branch", "main");
```

Comment:
Technically, `main` is our default, but I think we still allow alternative main branches. Could you check this?

Reply:
looking through the code it seems that main is hardcoded pretty much everywhere, `ProgrammingExerciseFactory` even has a constant `DEFAULT_BRANCH` which is set to main

Reply:
nevermind, I found a value in the configuration file, using that one now

Reply:
oh ok, so I should probably pass the default branch of the exercise all the way through to this method, got it. thanks!

Reply:
should be fixed now